### PR TITLE
Stabilise, Add Instructions, Prevent Crashing and Improve Accuracy of AR

### LINKED ISSUE #159: Fix Instructions on how to use the AR
User testing has revealed that AR instructions are unclear. Break instructions down into steps and show visual cues.

### PR DESCRIPTION
**Pull request by artur-varosyan**
This pull request brings in the following:

Additions & Changes:
* Instructions have been improved, with a mascot guiding the user in steps (change requested by user feedback, see #131) 
* Compass readings are now taken at a specified interval, and a median is taken for stabilisation
* Compass is polled in a separate thread
* Several products can be displayed at one time, with a scrollable view on the top
* Calculations surrounding the placement of AR objects have been improved, which means that objects are placed more accurately in the virtual space

Bugs fixed:
* Fix bug where incorrect usage of 3d models and shaders would result in a sudden crash
* Fix bug where going to a product page would result in a crash

Please let me know what you think!

Closes #159

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -52,13 +50,12 @@
...
-<!--            android:exported="true" />-->
-<!--        <activity-->
-<!--            android:name=".ProfileActivity"-->
-<!--            android:exported="true" />-->
+        </activity> <!-- <activity -->
+        <!-- android:name=".ChatListActivity" -->
+        <!-- android:exported="true" /> -->
+        <!-- <activity -->
+        <!-- android:name=".ProfileActivity" -->
+        <!-- android:exported="true" /> -->
```
Comment (Lingtao-Cai) [inline app/src/main/AndroidManifest.xml:58]: Hey Artur, could you remove those commented codes since they are not activity now

**Review by Lingtao-Cai — APPROVED** [review]
Great job!!!

**Review by arfi2212 — APPROVED** [review]
This makes AR so much better, thanks Artur! :)

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (artur-varosyan) [reply app/src/main/AndroidManifest.xml]: Done!

**Pull request merged**